packages/discord.js/typings/index.d.ts

```diff
@@ -2262,13 +2299,53 @@ export class SelectMenuInteraction<Cached extends CacheType = CacheType> extends
     SelectMenuComponent | APISelectMenuComponent,
     SelectMenuComponent | APISelectMenuComponent
   >;
-  public componentType: ComponentType.SelectMenu;
+  public componentType: ComponentType.StringSelect |
+   ComponentType.UserSelect |
+   ComponentType.RoleSelect |
+   ComponentType.MentionableSelect|
+   ComponentType.ChannelSelect;
   public values: string[];
   public inGuild(): this is SelectMenuInteraction<'raw' | 'cached'>;
   public inCachedGuild(): this is SelectMenuInteraction<'cached'>;
   public inRawGuild(): this is SelectMenuInteraction<'raw'>;
 }
-
+export class RoleSelectMenuInteraction<Cached extends CacheType = CacheType> extends SelectMenuInteraction<Cached> {
+  public componentType: ComponentType.RoleSelect;
+  public roles: Collection<Snowflake, Cached extends 'raw' ? APIRole : Role>;
```

Comment:
Use `CacheTypeReducer`
```suggestion
  public roles: Collection<Snowflake, CacheTypeReducer<Cached, Role, APIRole>>;
```
The same for others

Reply:
no 
CacheTypeReducer can be null when not cached
but it cant

Reply:
https://github.com/discordjs/discord.js/blob/6f446b9e6a528a8dc551620c9e7823d20db5e722/packages/discord.js/src/structures/RoleSelectMenuInteraction.js#L17

Reply:
You should use it anyway. At least like this.
https://github.com/discordjs/discord.js/blob/89a0ff9783a186461d23c9e25429b49b834ac332/packages/discord.js/typings/index.d.ts#L2298-L2304

Reply:
my is shorter

Reply:
But wrong. It doesn't check if the type is `'cached' | 'raw'`.

Reply:
it is working
no CacheType => Object | APIObject
raw => APIObject
cached => Object